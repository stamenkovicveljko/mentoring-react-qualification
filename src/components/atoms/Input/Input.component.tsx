import React, { useState } from "react";
import * as Styled from "./Input.styles";

interface Props extends React.HTMLAttributes<HTMLInputElement> {
  onChange?: (ev: React.ChangeEvent<HTMLInputElement>) => void;
  value?: string;
  placeholder?: string;
}

const Input = (props: Props) => {
  const { value = "", onChange = () => {}, placeholder, ...rest } = props;
  const [stateValue, setStateValue] = useState("");

  return (
    <Styled.Container>
      <input
        {...rest}
        value={value || stateValue}
        onChange={(ev) => {
          onChange(ev);
          setStateValue(ev.target.value);
        }}
        placeholder={placeholder}
      />
    </Styled.Container>
  );
};

export default Input;
