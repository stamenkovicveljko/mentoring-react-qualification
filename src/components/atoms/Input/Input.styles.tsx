import styled from "styled-components";
import { fontSize } from "../../../styles";
import { resetInput } from "../../../styles/resets";

export const Container = styled.div`
  max-width: 100%;
  width: 100%;
  > input {
    ${resetInput}
    ${fontSize("16px", "20px")}
    width: 100%;
    border-radius: 6px;
    padding: 10px;
  }
`;
