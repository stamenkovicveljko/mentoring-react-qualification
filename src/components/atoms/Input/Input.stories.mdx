import { Meta, Story, Preview, Props } from "@storybook/addon-docs/blocks";
import Input from "./Input.component";

<Meta title="Input" component={Input} />

# Input Component

This is a pretty straight forward Input that comes with the variants sufficient for the project.

## Implementation

<Preview>
  <Story name="default">
    <Input placeholder="Basic input with placeholder" />
  </Story>
</Preview>

The Input component can work by itself but most likely you'll need `value` and `onChange` props.

Example usage:

```tsx
const [inputValue, setInputValue] = useState('');

function handleInput(ev) {
    setInputValue(ev.target.value);
}

render(){
   return (
    /*...*/
    <Input value={valueInput} onChange={handleInput} />
    /*...*/
   )
}
```

## Props

<Props of={Input} />
