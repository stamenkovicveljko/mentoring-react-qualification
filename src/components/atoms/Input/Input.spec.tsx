import React from "react";
import renderer from "react-test-renderer";
import Input from "./Input.component";

test("Input renders as expected", () => {
  const component = renderer.create(<Input placeholder="test input" />);

  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
