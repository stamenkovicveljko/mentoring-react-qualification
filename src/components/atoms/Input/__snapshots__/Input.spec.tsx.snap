// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders as expected 1`] = `
<div
  className="sc-AxjAm jQZEFe"
>
  <input
    onChange={[Function]}
    placeholder="test input"
    value=""
  />
</div>
`;
